File: native-modules-playground/animating-numbers-expo-module/ios/AnimatingNumbersExpoModule.swift
import ExpoModulesCore

public class AnimatingNumbersExpoModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AnimatingNumbersExpoModule")

    View(AnimatingNumbersExpoModuleView.self) {
      Prop("value") { (view: AnimatingNumbersExpoModuleView, value: Double) in
        view.model.value = value
      }
    }
  }
}

File: native-modules-playground/animating-numbers-expo-module/android/src/main/java/expo/modules/animatingnumbersexpomodule/AnimatingNumbersExpoModule.kt
package expo.modules.animatingnumbersexpomodule

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class AnimatingNumbersExpoModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("AnimatingNumbersExpoModule")

    View(AnimatingNumbersExpoModuleView::class) {
      Prop("value") { view: AnimatingNumbersExpoModuleView, value: Double ->
        view.value = value
      }
    }
  }
}
